### ISSUE TITLE
Hash in code snippets gets modified

### ISSUE THREAD

**Issue by gfwilliams** [microcosm-cc/microweb#288]
Hi,

There's a bug where hashes in code snippets get converted to markdown for links (I guess it's an over-enthusiastic regex). For instance

```
#define foo
```

Will go to

```
[#define](http://forum.espruino.com/search­/?q=%23define) foo
```

Finally, it's not really a bug, but most of my users don't get that they need an empty newline before and after the 3 backticks to make the code thing work and I end up having to manually edit their post. I wonder if it's possible to somehow auto-enforce that?

**Comment by buro9** [comment]
That's actually Twitter hashtag support within comments.

Which has proved a subtle feature that gets ignored for a while and then gets used quite well.

This could also be a site option.

**Comment by gfwilliams** [comment]
Yep, I gathered that - some folks have been using it already, so it'd be a shame to see it go.

The problem is when it's in code snippets though - because then you really do get the text `[#define](http://forum.espruino.com/search­/?q=%23define) foo` inserted into the middle of your code (rather than just a link).

**Comment by buro9** [comment]
When I'm back in London I'll look and seehow we find the hashtags. It could be something I make contextual and avoid within pre and code blocks that get generated from markdown.

**Comment by gfwilliams** [comment]
Thanks! 

There's no rush on this or the other bug though - I just posted up because you'd checked on the Espruino site :)

**Comment by buro9** [comment]
We did CNAME changes at CloudFlare yesterday, I was checking all sites. And then it occurred to me that maybe you had issues I didn't know about since there had been silence for so long.
